.github/workflows/reusable_build.yml

```diff
@@ -90,9 +90,12 @@ jobs:
         run: |
           if [[ "${{ needs.call_get_app_metadata.outputs.is_rust }}" == "true" ]];
           then
-              DEVICE_NAME="$(echo ${{ matrix.device }} | sed 's/nanosp/nanosplus/')" && \
+              BUILD_DEVICE_NAME="$(echo ${{ matrix.device }} | sed 's/nanosp/nanosplus/')" && \
+              BIN_DIR_NAME="$(echo ${{ matrix.device }} | sed 's/nanosp/nanos2/')" && \
               cd ${{ needs.call_get_app_metadata.outputs.build_directory }} && \
-              cargo ledger build ${DEVICE_NAME} -- -Zunstable-options --out-dir=./build/${{ matrix.device }}/bin/
+              cargo ledger build ${BUILD_DEVICE_NAME} -- -Zunstable-options --out-dir=./build/${BIN_DIR_NAME}/bin/
+              APP_NAME=$(grep -E '^\s*name\s*=\s*' Cargo.toml | awk -F'["]' '{print $2}')
+              mv ./build/${BIN_DIR_NAME}/bin/${APP_NAME} ./build/${BIN_DIR_NAME}/bin/app.elf
```

Comment:
Would be great if we can remove this part at some point! But let's keep it for another time.